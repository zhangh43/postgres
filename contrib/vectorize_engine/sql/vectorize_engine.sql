--
-- Regression Tests for Custom Plan APIs
--

-- construction of test data

CREATE TABLE t1 (
    a   int,
	b	double precision
);
INSERT INTO t1 SELECT generate_series(1,3), 2.3;
INSERT INTO t1 SELECT generate_series(1,3), 3.3;
INSERT INTO t1 SELECT generate_series(1,3), 4.3;
VACUUM ANALYZE t1;

create extension vectorize_engine;
SET enable_vectorize_engine TO on;
SELECT * FROM t1;
SELECT b FROM t1;
SELECT b+1 FROM t1;
SELECT count(b) FROM t1;
SELECT a, sum(b), avg(b)  FROM t1 group by a;
SELECT a, sum(b), avg(b)  FROM t1 where a < 3 group by a;


drop extension vectorize_engine;
