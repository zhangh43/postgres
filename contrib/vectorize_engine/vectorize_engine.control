# vectorize_engine extension
comment = 'vectorize_engine'
default_version = '1.0'
module_pathname = '$libdir/vectorize_engine'
relocatable = true
