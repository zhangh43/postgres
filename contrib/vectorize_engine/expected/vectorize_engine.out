--
-- Regression Tests for Custom Plan APIs
--
-- construction of test data
CREATE TABLE t1 (
    a   int,
	b	double precision
);
INSERT INTO t1 SELECT generate_series(1,3), 2.3;
INSERT INTO t1 SELECT generate_series(1,3), 3.3;
INSERT INTO t1 SELECT generate_series(1,3), 4.3;
VACUUM ANALYZE t1;
create extension vectorize_engine;
SET enable_vectorize_engine TO on;
SELECT * FROM t1;
 a |  b  
---+-----
 1 | 2.3
 2 | 2.3
 3 | 2.3
 1 | 3.3
 2 | 3.3
 3 | 3.3
 1 | 4.3
 2 | 4.3
 3 | 4.3
(9 rows)

SELECT b FROM t1;
  b  
-----
 2.3
 2.3
 2.3
 3.3
 3.3
 3.3
 4.3
 4.3
 4.3
(9 rows)

SELECT b+1 FROM t1;
 ?column? 
----------
      3.3
      3.3
      3.3
      4.3
      4.3
      4.3
      5.3
      5.3
      5.3
(9 rows)

SELECT count(b) FROM t1;
 count 
-------
     9
(1 row)

SELECT a, sum(b), avg(b)  FROM t1 group by a;
 a | sum | avg 
---+-----+-----
 1 | 9.9 | 3.3
 2 | 9.9 | 3.3
 3 | 9.9 | 3.3
(3 rows)

SELECT a, sum(b), avg(b)  FROM t1 where a < 3 group by a;
 a | sum | avg 
---+-----+-----
 1 | 9.9 | 3.3
 2 | 9.9 | 3.3
(2 rows)

drop extension vectorize_engine;
